Source/WebCore/dom/CallbackResult.h

```diff
@@ -39,23 +40,29 @@ template<typename ReturnType> class CallbackResult {
 public:
     CallbackResult(CallbackResultType);
     CallbackResult(ReturnType&&);
+    CallbackResult(JSC::Exception*);
 
     CallbackResultType type() const;
     ReturnType&& releaseReturnValue();
+    JSC::Exception& exceptionObject() const;
 
 private:
     Expected<ReturnType, CallbackResultType> m_value;
+    JSC::Exception* m_exception = nullptr;
 };
 
 template<> class CallbackResult<void> {
 public:
     CallbackResult() = default;
     CallbackResult(CallbackResultType);
+    CallbackResult(JSC::Exception*);
 
     CallbackResultType type() const;
+    JSC::Exception& exceptionObject() const;
 
 private:
     CallbackResultType m_type = CallbackResultType::Success;
+    JSC::Exception* m_exception = nullptr;
```

Comment:
Could use `NakedPtr<JSC::Exception>`, then you don't need explicit null initialization.